### ISSUE TITLE
Errors on adding folder or file collection

### ISSUE THREAD

**Issue by wini2** [xperseguers/t3ext-file_list#53]
Same Problem as described in #42 
Trying to add a folder, I get an error in the modal window

503
Oops, an error occurred!

PHP Warning: Undefined array key "" in /home/www/cms54/vendor/typo3/cms-backend/Classes/Controller/AbstractLinkBrowserController.php line 392.

Adding the path manually works but leads to an error in FE.

When I use a File collection the FE also shows Oops, an error occurred! 

[TYPO3 12.4.11/filelist  3.0.1]

**Comment by xperseguers** [comment]
Juste tested with TYPO3 12.4.11, mode is set to FOLDER (files within a FOLDER). When I choose to restrict to some folder path, both icons on the right of the textbox eye/anchor and anchor link just work fine.

**Comment by xperseguers** [comment]
This is how it looks like:

<img width="996" alt="image" src="https://github.com/xperseguers/t3ext-file_list/assets/646805/3c9e432b-30c3-444d-9b11-c4a359613622">

If you have more items pointed by the arrow, please double check the corresponding extensions as the problem may come from them instead.

**Comment by wini2** [comment]
I don't use ext. that make changes to the Link Browser. My Link Browser dont't show up

![2024-02-26_14-39-37](https://github.com/xperseguers/t3ext-file_list/assets/6897742/7787cdf1-630b-468a-9628-45eaea66a316)
clicking the link-Button ends up with an error
![2024-02-26_14-41-20](https://github.com/xperseguers/t3ext-file_list/assets/6897742/7cd7c696-ff0c-4e57-9e84-21f93dd6a852)

**Comment by xperseguers** [comment]
Line 392 of that file is:

```
$this->displayedLinkHandler = $this->linkHandlers[$this->displayedLinkHandlerId]['handlerInstance'];
```

(as seen on https://github.com/TYPO3/typo3/blob/12.4/typo3/sysext/backend/Classes/Controller/AbstractLinkBrowserController.php#L392)

This means that `$this->displayedLinkHandlerId` is empty.

The line before that variable is supposed to get initialized with `(string)array_key_first($menuDef)`. This means that somehow your `$menuDef` is strange.

In my test install, I have a few other extensions but here is what it contains:

<img width="384" alt="image" src="https://github.com/xperseguers/t3ext-file_list/assets/646805/108fa93f-7cf3-40e3-83c0-a3781eba61ee">

Can you please debug what you get at that point?

**Comment by xperseguers** [comment]
What you could do as well to see if somehow it helps is changing line 386 from

```php
if (!$this->displayedLinkHandler) {
```

to

```php
if (!$this->displayedLinkHandler && !empty($menuDef)) {
```

This should prevent the error and show the popup although I doubt you will have any folder to choose from.

And I still struggle at understanding 1) why it happens to you and 2) what it have to do with EXT:file_list.

**Comment by wini2** [comment]
Changing the code in AbstractLinkBrowserController.php leads to a less specific error message 
Call to a member function modifyLinkAttributes() on null 

I'm not familiar with PHP and I dont't know how to debug the menu like you proposed me to do or how to change the TCA/flexform-settings to nativ-folder-browser.

**Comment by xperseguers** [comment]
> I'm not familiar with PHP and I dont't know how to debug the menu like you proposed me to do or how to change the TCA/flexform-settings to nativ-folder-browser.

That's not PHP specific.

How did you install this extension? Using composer? If so edit your `composer.json` and change the requirement to file_list to this:

```
"causal/file_list": "dev-feature/53-native-folder-browser"
```

then run `composer update`.

If you cloned with git manually, go to the file_list directory and do

```
cd /path/to/vendor/causal/file_list
git checkout feature/53-native-folder-browser
```

As a last resort, replace the whole content of `file_list` with the zip you can get from https://github.com/xperseguers/t3ext-file_list/tree/feature/53-native-folder-browser when you click on the big green "Code" button.

**Comment by wini2** [comment]
Thanks for your hint.
After the update (with the original AbstractLinkBrowserController.php), the appearence for file_collections changed and I can choose and save the selection, but no change for Folder. 
![2024-02-26_18-59-45](https://github.com/xperseguers/t3ext-file_list/assets/6897742/ea60b8d1-2f04-4980-ac9f-2cc4f457906f)

In the FE I get the following error
Causal\FileList\Controller\FileController ERROR:

The configuration of the file_list plugin is incorrect: Causal\FileList\Domain\Repository\FileRepository::getFolderByIdentifier() expects a FAL identifier. Input was: /fileadmin/user_upload/0000-Musterbilder/

**Comment by xperseguers** [comment]
> After the update (with the original AbstractLinkBrowserController.php), the appearence for file_collections changed and I can choose and save the selection, but no change for Folder.

I don't get what you mean with "the original AbstractLinkBrowserController.php" as what I expect now to be tested, with TYPO3 v12, is the other branch I created specifically for TYPO3 v12 (and soon v13).

From the screenshot, obviously the branch feature/53-native-folder-browser is not being used, or you forgot to clear all caches?

For the other error, `/fileadmin/user_upload` is indeed not a proper identifier, but since the folder selector is obviously not the correct one, there's no point discussing this.

**Comment by wini2** [comment]
Original AbstractLinkBrowserController.php means, I made the change
if (!$this->displayedLinkHandler && !empty($menuDef)) {
undone.
I use TYPO3 12.4.11 with file_list 3.01 
For the update I have used "causal/file_list": "dev-feature/53-native-folder-browser" setting, ran composer update causal/file_list and cleared all caches.

**Comment by xperseguers** [comment]
That's how the folder field is supposed to look in TYPO3 v12 if you are on the proper branch:

<img width="666" alt="image" src="https://github.com/xperseguers/t3ext-file_list/assets/646805/22ef9a9e-4141-4cf0-a91c-9b95178011b0">

**Comment by xperseguers** [comment]
Published version 3.1.0. You can update from there and that will fix that problem...

**Comment by wini2** [comment]
After updating to 3.1.0 it works like expected.
 
Big Thanks for your dedicated help.
